pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -325,3 +336,34 @@ func isBackupConditionExist(conditions []configv1.ClusterOperatorStatusCondition
 	}
 	return false
 }
+
+func getCurrentClusterVersion(cv *configv1.ClusterVersion) (string, error) {
+	if cv == nil {
+		return "", fmt.Errorf("ClusterVersion type is nil: %v", cv)
+	}
+	for _, c := range cv.Status.History {
+		if c.State == configv1.CompletedUpdate {
+			return c.Version, nil
+		}
+	}
+	return "", fmt.Errorf("unable to retrieve cluster version, no completed update was found in status history: %v", cv.Status.History)
+}
+
+func isNewBackupRequired(config *configv1.ClusterVersion, clusterOperatorStatus *configv1.ClusterOperatorStatus) (bool, error) {
+	currentVersion, err := getCurrentClusterVersion(config)
+	if err != nil {
+		return false, err
+	}
+
+	backupCondition := configv1helpers.FindStatusCondition(clusterOperatorStatus.Conditions, backupConditionType)
+	if backupCondition == nil {
+		return false, fmt.Errorf("clusterOperatorStatus %v does not contain %v type", clusterOperatorStatus.Conditions, backupConditionType)
+	}
+	if backupCondition.Reason == backupSuccess {
+		backupVersion := strings.Fields(backupCondition.Message)[2]
+		if cmp := version.CompareSimple(currentVersion, backupVersion); cmp > 0 {
```

Comment:
so if we were initially on 4.9, then upgraded to 4.10, then we have a backup for 4.9.
Now we want to upgrade again for 4.11, and currently on 4.10. 

The current version will not be in the message, and we still do not create a backup for 4.10 before upgrading 

Please correct me if i am wrong

Reply:
> so if we were initially on 4.9, then upgraded to 4.10, then we have a backup for 4.9.
> Now we want to upgrade again for 4.11, and currently on 4.10.
> The current version will not be in the message, and we still do not create a backup for 4.10 before upgrading

on the second upgrade you're on 4.10, so you would check whether you have a completed backup for 4.10 (which you haven't), you only have one for 4.9. Or am I missing anything?

Reply:
that also brings up some questions around downgrades and subsequent upgrades, could you also leave a comment there about how this case is handled and why it is fine? basically what Trevor mentioned in the BZ

Reply:
So I figured out the issue with current code, it triggers the update once a condition on CV `ReleaseAccepted` == `false` is found, and that is fine and works. 

The issue arise with the consecutive upgrades, because the condition is true and is never false again. 
This PR adds another condition bases on CEO condition && CV upgrade history. 

Once we find an upgrade with `partial` state (i.e. in progress ) we check the `RecentBackup` condition on CEO, and from it we know the version of the most recent backup, then we decide if we need a new backup or we are good